File: frontend/src/app/api/models/update-expense-dto.ts
/* tslint:disable */
import { ExpenseStatus } from './expense-status';
export interface UpdateExpenseDto {
  adnotation?: string;
  amount: number;
  deposit: number;
  expenseStatus: ExpenseStatus;
  id: number;
  name: string;
  plannerId: number;
  provider: string;
  serviceProviderId?: number;
}

File: WPA.backend/WPA.backend/DTOs/Expenses/UpdateExpenseDto.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using WPA.backend.Entities;

namespace WPA.backend.DTOs.Expenses
{
    public class UpdateExpenseDto
    {
        [Required]
        public int Id { get; set; }

        [Required]
        public int PlannerId { get; set; }

        public int? ServiceProviderId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public double Amount { get; set; }

        public string Adnotation { get; set; }

        [Required]
        public double Deposit { get; set; }

        [Required]
        public ExpenseStatus ExpenseStatus { get; set; }
    }
}
